keep/step/step.py

```diff
@@ -51,40 +51,50 @@ def _post_step_validations(self):
         self.logger.debug("Post step validation")
         foreach = self.step_config.get("foreach")
         if foreach:
-            self._post_each_step_validations(foreach.get("value"))
+            self._post_each_step_validations(foreach)
         else:
             self._post_single_step_validations()
         self.logger.debug("Post Step validation success")
 
     def _post_each_step_validations(self, foreach_value_template):
-        context = self.context_manager.get_full_context()
         foreach_actual_value = self._get_actual_value(foreach_value_template)
         for value in foreach_actual_value:
             # will be use inside the io handler
             self.context_manager.set_for_each_context(value)
             for condition in self.step_conditions:
                 condition_type = condition.get("type")
                 condition_alias = condition.get("alias")
-                condition = ConditionFactory.get_condition(condition_type, condition)
+                condition_name = (
+                    condition.get("name", None) or condition_type
+                )  # if name is not supplied, its ok to use the type as the name (todo: this is a hack, we should fix this)
```

Comment:
let's maybe fix that now? make it a "breaking change", otherwise I feel like the hole will just get deeper, wdyt?

Reply:
I think this should be implicit until we have e2e tests, then it'll be easy to fix all underlying alerts.